### PR TITLE
[Wait for #3925] Validate CausalLM Models and Add Model-Level Unittest

### PR DESCRIPTION
**Pull request by baek2sm**
This PR adds model-level validation coverage for CausalLM models.

- Qwen2.5  / Qwen2.5 Embedding
- Qwen3 / Qwen3 Embedding
- Gemma3 / Gemma3 Embedding

validated with python logits and made tiny unit-tests for these models.

Signed-off-by: SeungBaek Hong <sb92.hong@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM
